lua/nvim-tree/diagnostics.lua

```diff
@@ -170,8 +170,13 @@ function M.setup(opts)
 
   if M.enable then
     log.line("diagnostics", "setup")
-    vim.cmd "au DiagnosticChanged * lua require'nvim-tree.diagnostics'.update()"
-    vim.cmd "au User CocDiagnosticChange lua require'nvim-tree.diagnostics'.update()"
+    a.nvim_create_autocmd("DiagnosticChanged", {
+      callback = M.update,
+    })
+    a.nvim_create_autocmd("User", {
+      pattern = "CocDiagnosticChanged",
+      callback = M.update,
+    })
```

Comment:
DiagnosticChanged works, CocDiagnosticChanged does not.

See https://github.com/kyazdani42/nvim-tree.lua/pull/1218#issuecomment-1114469577 for a test harness you can use.

Reply:
@alex-courtis I fix a typo, should work now